src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -2030,17 +2000,17 @@ private static void ImportKnownTypeAttributes(Type? type, Dictionary<Type, Type>
                     }
                 }
 
+                // NOTE TODO smolloy - This entire try/catch block is new in CoreFx from the beginning. I wonder why it wasn't there in NetFx. Is it needed there? Or not-needed here?
```

Comment:
>                // NOTE TODO smolloy - This entire try/catch block is new in CoreFx from the beginning. I wonder why it wasn't there in NetFx. Is it needed there? Or not- [](http://example.com/codeflow?start=0&length=170)

Isn't the answer in the catch block comment?

Reply:
But if I believe the catch block... shouldn't this also be in NetFx? Why isn't it?

Reply:
I think the comment is slightly wrong and is actually supposed to mention the Serializable attribute as it mentions KeyValuePair. In NetFx it's attributed, but in Core/.NET it isn't. That's why it's not applicable to NetFx

Reply:
Browsing through our project board, I see you also noticed this a couple of months ago. #67949

Reply:
// After careful consideration, I don't think this code is necessary anymore. After trying to
                // decipher the intent of the comments here, my best guess is that this was DCJS's way of working
                // around a non-[Serializable] KVP in Silverlight/early .Net Core. The regular DCS went with a
                // KVPAdapter approach. Neither is needed now.
                //
                // But this code does produce additional artifacts in schema handling. To be cautious, just in case
                // somebody needs a KVP contract in addition to the KV contract in their schema, I've kept this
                // here behind this AppContext switch.